### PR TITLE
profiles

### PR DESCRIPTION
**Pull request by saml33**
adds profile features

### CODE REVIEW

File: public/locales/zh/profile.json [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
     "profile-pic-remove-success": "删除头像成功",
+    "profile-update-fail": "Failed to update profile",
+    "profile-update-success": "Profile updated",
     "remove": "删除",
-    "set-profile-pic": "设置头像"
-}
\ No newline at end of file
+    "save-profile": "Save Profile",
+    "set-profile-pic": "设置头像",
+        "swing-trader": "Swing Trader",
```
Comment (mschneider) [inline public/locales/zh/profile.json:32]: fix fmt

File: pages/profile.tsx [hunk 2]
```diff
@@ -0,0 +1,930 @@
...
+  onClose: () => void
+  profile?: any
+}) => {
+  const { t } = useTranslation(['common', 'profile'])
+  const { publicKey, signMessage } = useWallet()
+  const [profileName, setProfileName] = useState(
+    startCase(profile?.profile_name) || ''
+  )
+  const [traderCategory, setTraderCategory] = useState(
+    profile?.trader_category || TRADER_CATEGORIES[0]
```
Comment (mschneider) [inline pages/profile.tsx:657]: should we choose a default? people might find this awkward

Comment (saml33) [reply pages/profile.tsx]: The api applies a default of "Trader". Can ask Nick to remove it if you want.

File: pages/profile.tsx [hunk 3]
```diff
@@ -0,0 +1,930 @@
...
+  const [initialFollowingLoad, setInitialFollowingLoad] = useState(false)
+  const [loadProfileDetails, setLoadProfileDetails] = useState(false)
+  const [profilePk, setProfilePk] = useState<string | string[]>('')
+  const mangoGroup = useMangoStore((s) => s.selectedMangoGroup.current)
+  const mangoClient = useMangoStore((s) => s.connection.client)
+  const mangoCache = useMangoStore((s) => s.selectedMangoGroup.cache)
+  const mangoAccounts = useMangoStore((s) => s.mangoAccounts)
+  const { publicKey, connected, wallet } = useWallet()
+  const router = useRouter()
+  const { pk } = router.query
```
Comment (riordanp) [inline pages/profile.tsx:82]: Is it possible to (also?) check for 'pubkey' as a query parameter? Just for consistency with the account page URL structure.

File: pages/profile.tsx (hunk 2 above)
Comment (mschneider) [reply pages/profile.tsx]: guess the first category in that array should then be "Trader" as well

Comment (saml33) [reply pages/profile.tsx]: The first one is "Day Trader" for me. It's alphabetical. I changed the fallback to TRADER_CATEGORIES[5] the other day which is "Trader"

File: pages/profile.tsx (hunk 3 above)
Comment (saml33) [reply pages/profile.tsx:82]: Had a look. It needs to be different from the account query as that is used for something else in the topbar. We are going to change it to use the profile name as they have enforced uniqueness and are more user-friendly.

**Review by mschneider — APPROVED** [review]

**Pull request merged**